contracts/examples/PairFlash.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import '@uniswap/v3-core/contracts/interfaces/callback/IUniswapV3FlashCallback.sol';
+import '@uniswap/v3-core/contracts/libraries/LowGasSafeMath.sol';
+
+import '../base/PeripheryPayments.sol';
+import '../base/PeripheryImmutableState.sol';
+import '../libraries/PoolAddress.sol';
+import '../libraries/CallbackValidation.sol';
+import '../libraries/TransferHelper.sol';
+import '../interfaces/ISwapRouter.sol';
+
+/// @title Flash contract implementation
+/// @notice An example contract using the Uniswap V3 flash function
+contract PairFlash is IUniswapV3FlashCallback, PeripheryImmutableState, PeripheryPayments {
+    using LowGasSafeMath for uint256;
+    using LowGasSafeMath for int256;
+
+    ISwapRouter swapRouter;
```

Comment:
this could be immutable (and maybe public too...might be helpful when integrating)

Reply:
nvm...maybe just public, I guess it can't be immutable since it's an interface!